extensions/who-is-visiting-this-content/app.R

```diff
@@ -177,38 +314,145 @@ server <- function(input, output, session) {
 
     filtered_visits |>
       left_join(unfiltered_hits, by = "user_guid") |>
-      left_join(user_names(), by = "user_guid") |>
-      replace_na(list(full_name = "[Anonymous]")) |>
+      left_join(users(), by = "user_guid") |>
+      replace_na(list(display_name = "[Anonymous]")) |>
       arrange(desc(n_visits)) |>
-      select(n_visits, n_hits, full_name, username)
+      select(user_guid, display_name, email, n_visits)
   })
 
-  summary_message <- reactive({
-    content_title <- content() |>
-      filter(guid == input$content_guid) |>
-      pull(title)
+  selected_content_info <- reactive({
+    filter(content(), guid == input$guid_field)
+  })
+
+  # Create day by day hit data for plot
+  daily_hit_data <- reactive({
+    all_dates <- seq.Date(date_range()[1], date_range()[2], by = "day")
+
+    all_visits_data() |>
+      mutate(date = date(timestamp)) |>
+      group_by(date) |>
+      summarize(daily_visits = n(), .groups = "drop") |>
+      tidyr::complete(date = all_dates, fill = list(daily_visits = 0))
+  })
```

Comment:
If we aggregate at the day level when we plot (have you tried using `scale_x_date(date_breaks = "days")` https://ggplot2.tidyverse.org/reference/scale_date.html#arg-date-breaks ?), that might help us in making the more detailed plot where we see over time since we're not collapsing the data until we plot it.

Reply:
I had already been using `scale_x_datetime()` in the new plot, and this prompted me to also add `scale_x_date()` to the old plot, which looks nicer, but doesn't work for longer date ranges.

I tried to just specify minor breaks to at least have the nice visual rhythm provided by daily lines, but ggplotly does not respect minor breaks

https://github.com/plotly/plotly.R/issues/952

Reply:
Going to try calculating the ticks by like, looking at the total number of days.

Reply:
_nods_ yeah I always get tripped up by the ways ggplotly doesn't totally work. I think it's ok to leave this as is now that we have the timeline view especially. It would be nice to be able to be pure(r) in the separation between our data visualization code and our data munging, but it's ok if we can't for other reasons like this

Reply:
I'll leave this convo unresolved for now, since we _would_ like to be able to have better ticks.

At least mousing over a column tells you what date it is.